test/modules/core/shadertools/pack.spec.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015 - 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the 'Software'), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {Buffer, Transform, registerShaderModules} from 'luma.gl';
+import test from 'tape-catch';
+import {default as pack} from '../../../../modules/imageprocessing/src/modules/pack';
+import {fixture} from 'luma.gl/test/setup';
+// TODO: test WebGL1 when Transform class is ready
+const gl = fixture.gl2;
+
+const EPSILON = 1e-5;
+
+// failure when 1e-6
+// expected: 100.54000091552734 actual: 100.54000854492188 diff: 0.00000762939453125
+
+// failure when 1e-7
+// expected: 1.2640000581741333 actual: 1.2640001773834229 diff: 1.1920928955078125e-7
+
+registerShaderModules([pack]);
```

Comment:
Shader modules are designed to be used without registration. They can be supplied directly as part of `modules` array.